### PR TITLE
[MEI-56]: Add prose and mdx linting

### PR DESCRIPTION
**Pull request by justkahdri**
# Pull Request

## Related issue
Fixes https://github.com/meilisearch/documentation/issues/2239, https://github.com/meilisearch/documentation/issues/2291

## What does this PR do?
- Re-adds Vale
- Re-adds markdownlint

## PR checklist
- [ ] Does it works as expected from the meilisearch team?
- [ ] Is `eslint` necessary? Do we need to lint custom components?

### CODE REVIEW

**Comment by guimachiavelli** [comment]
It took me some time, but I finally have the time to follow up on this.

I ran both linters locally, then picked ten or so warning/errors and they are all valid and in the correct line. I will submit fixes for the indicated issues.

The only problem I had so far has been with the `yarn proselint` command not scanning folders with more than one level of nesting. For example, running `yarn proselint` returns 13 suggestions on `resources/faq.mdx`, but running `vale ./*/*/*.mdx` returns, well a lot more. Is there any globbing pattern that will include all `.mdx` files, no matter their nesting level?

**Comment by justkahdri** [comment]
> Is there any globbing pattern that will include all .mdx files, no matter their nesting level?

Hmm, I'm not sure. I tried different wildcards (*) combinations and they don't work as expected for `vale`, how would you run it for md files? Maybe I'm missing some config to treat `mdx` as `md` entirely

**Comment by guimachiavelli** [comment]
@justkahdri, I've done some digging and am actually not sure how we got it to work in the previous website. We ran `vale .` and things worked out fairly well in terms of globbing.

I've had some limited success with changing `vale.ini` L#9 from `[*]` to  `[*.md]`, then running `vale .`. This still includes all content inside the `.github` dir, but perhaps we can use `--glob` to prevent that from happening? Alternatively, we could also explicitly list the directories with content, though that's a liable to break in the future if/when we create a new top-level content dir.

**Comment by justkahdri** [comment]
@guimachiavelli sadly, that's how the output is formatted by the original [markdownlint](https://github.com/DavidAnson/markdownlint) tool. I couldn't find any optional flag or format style for the output in their README. The only option available is to print the result in a specific file or show the output in json format, which I don't think that makes it any better 😓

Screenshot 2024-01-18 at 11 12 53

File: .markdownlint.jsonc [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "MD026": {"punctuation": ".,;:。，；"}, // no-trailing-punctuation Trailing punctuation in headings except ? and ! 
+  "MD014": false, // commands-show-output Dollar signs used before commands without showing output
+  "MD024": false, // no-duplicate-heading/no-duplicate-header
+  "MD025": {"front_matter_title": "(?!.*)"}, // single-title/single-h1 Accept both h1 and front matter titles
+  "MD040": false, // fenced-code-language Fenced code blocks should have a language specified
+  "MD033": false, // no-inline-html Inline HTML
+  "MD049": { "style": "underscore" }, // enforce use of underscore for em/italics
+  "MD050": { "style": "asterisk" }, // enforce use of asterisk for strong/bold
+  "MD051": false
+}
```
Comment (guimachiavelli) [inline .markdownlint.jsonc:13]:
```suggestion
}

```

File: .markdownlintignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+node_modules
+.vuepress
+public
+README.md
```
Comment (guimachiavelli) [inline .markdownlintignore:4]:
```suggestion
README.md

```

File: .vale/styles/Meilisearch/Indexing.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+# Error: Meilisearch.indexing.yml
+
+# Most dictionaries still list "indexation" as a term with a very specific meaning in economy so we should use “indexing”
+extends: substitution
+message: "Use '%s' instead of '%s'."
+level: warning
+swap:
+  indexation: indexing
+
```
Comment (guimachiavelli) [inline .vale/styles/Meilisearch/Indexing.yml:10]:
```suggestion

```

File: .vale/styles/Meilisearch/MergeConflictMarkers.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+# Checks for the presence of merge conflict markers.
+extends: existence
+message: 'Merge conflict marker "%s" found.'
+link: https://docs.gitlab.com/ee/development/code_review.html#merging-a-merge-request
+level: error
+scope: raw
+raw:
+  - '\n<<<<<<< .+\n|\n=======\n|\n>>>>>>> .+\n'
+
```
Comment (guimachiavelli) [inline .vale/styles/Meilisearch/MergeConflictMarkers.yml:12]:
```suggestion

```

File: .vale/styles/Meilisearch/Repetition.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+# Error: Meilisearch.Repetition.yml
+
+# catch any instances of repeated words. E.g., "This is is a sentence"
+extends: repetition
+message: "'%s' is repeated."
+level: error
+alpha: true
+tokens:
+  - '[^\s]+'
```
Comment (guimachiavelli) [inline .vale/styles/Meilisearch/Repetition.yml:10]:
```suggestion
  - '[^\s]+'

```

File: .vale/styles/Meilisearch/Spacing.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+message: "'%s' should have one space."
+link: https://docs.microsoft.com/en-us/style-guide/punctuation/periods
+level: error
+nonword: true
+tokens:
+  - '[a-z]\s{2,}[a-z]' # ensure only one space between words
+  - '[a-z][.?!:;",] {2,}[A-Z]' # one space between sentences (end.  new)
+  - '[a-z][.?!:;",][A-Z]' #(end.new)
+  - '[\w.?!,\(\)\-":] {2,}[\w.?!,\(\)\-":]' #only one space after ,.?
+
```
Comment (guimachiavelli) [inline .vale/styles/Meilisearch/Spacing.yml:15]:
```suggestion

```

File: .vale/styles/Meilisearch/Indexing.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+# Error: Meilisearch.indexing.yml
+
+# Most dictionaries still list "indexation" as a term with a very specific meaning in economy so we should use “indexing”
+extends: substitution
+message: "Use '%s' instead of '%s'."
+level: warning
+swap:
+  indexation: indexing
```
Comment (guimachiavelli) [inline .vale/styles/Meilisearch/Indexing.yml:9]:
```suggestion
  indexation: indexing

```

File: .vale/styles/Meilisearch/MergeConflictMarkers.yml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+# Error: Meilisearch.MergeConflictMarkers.yml
+
+# Checks for the presence of merge conflict markers.
+extends: existence
+message: 'Merge conflict marker "%s" found.'
+link: https://docs.gitlab.com/ee/development/code_review.html#merging-a-merge-request
+level: error
+scope: raw
+raw:
+  - '\n<<<<<<< .+\n|\n=======\n|\n>>>>>>> .+\n'
```
Comment (guimachiavelli) [inline .vale/styles/Meilisearch/MergeConflictMarkers.yml:11]:
```suggestion
  - '\n<<<<<<< .+\n|\n=======\n|\n>>>>>>> .+\n'

```

File: .vale/styles/Meilisearch/Spacing.yml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+extends: existence
+message: "'%s' should have one space."
+link: https://docs.microsoft.com/en-us/style-guide/punctuation/periods
+level: error
+nonword: true
+tokens:
+  - '[a-z]\s{2,}[a-z]' # ensure only one space between words
+  - '[a-z][.?!:;",] {2,}[A-Z]' # one space between sentences (end.  new)
+  - '[a-z][.?!:;",][A-Z]' #(end.new)
+  - '[\w.?!,\(\)\-":] {2,}[\w.?!,\(\)\-":]' #only one space after ,.?
```
Comment (guimachiavelli) [inline .vale/styles/Meilisearch/Spacing.yml:14]:
```suggestion
  - '[\w.?!,\(\)\-":] {2,}[\w.?!,\(\)\-":]' #only one space after ,.?

```

**Review by guimachiavelli — APPROVED** [review]
LGTM 👍 

Should I merge it or are there any remaining steps you need to take care of on your side, @justkahdri?

**Comment by justkahdri** [comment]
We're good to go @guimachiavelli 🚀

**Pull request merged**